web-app/src/app/router/Router.tsx

```diff
@@ -17,8 +17,34 @@ import TermsAndConditions from '../screens/TermsAndConditions';
 import PrivacyPolicy from '../screens/PrivacyPolicy';
 import Feed from '../screens/Feed';
 import Feeds from '../screens/Feeds';
+import { SIGN_OUT_TARGET } from '../constants/Navigation';
+import {
+  LOGIN_CHANNEL,
+  LOGOUT_CHANNEL,
+  createDispatchChannel,
+} from '../services/channel-service';
+import { useAppDispatch } from '../hooks';
+import { logout } from '../store/profile-reducer';
 
 export const AppRouter: React.FC = () => {
+  const navigateTo = useNavigate();
+  const dispatch = useAppDispatch();
+  const logoutUser = (): void => {
+    dispatch(
+      logout({ redirectScreen: SIGN_OUT_TARGET, navigateTo, propagate: false }),
+    );
+  };
+
+  const loginUser = (): void => {
+    // Refresh the page to ensure the user is authenticated
+    window.location.reload();
+  };
+
+  useEffect(() => {
+    createDispatchChannel(LOGOUT_CHANNEL, logoutUser);
+    createDispatchChannel(LOGIN_CHANNEL, loginUser);
+  }, [dispatch]);
```

Comment:
how come we use `[dispatch]` over `[]` if we want these channels to be init once?

Reply:
The dispatch instance does not change between renders and guarantees that the app is in _ready_ state. I can switch it to _[]_, it have the same effect.

Reply:
Fixed!